olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "olp/dataservice/read/VersionedLayerClientImpl.h"
+
+#include <olp/core/cache/DefaultCache.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/core/context/Context.h>
+#include <olp/core/thread/TaskScheduler.h>
+#include "repositories/CatalogRepository.h"
+#include "repositories/DataRepository.h"
+#include "repositories/ExecuteOrSchedule.inl"
+#include "repositories/PartitionsRepository.h"
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+VersionedLayerClientImpl::VersionedLayerClientImpl(
+    olp::client::HRN catalog, std::string layer_id,
+    olp::client::OlpClientSettings client_settings)
+    : hrn_(std::move(catalog)),
+      layer_id_(std::move(layer_id)),
+      client_settings_(std::move(client_settings)) {
+  if (!client_settings_.cache) {
+    client_settings_.cache =
+        olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+  }
+}
+
+VersionedLayerClientImpl::~VersionedLayerClientImpl() = default;
+
+olp::client::CancellationToken VersionedLayerClientImpl::GetData(
+    DataRequest data_request, Callback callback) {
+  auto context = olp::client::CancellationContext();
+  olp::client::CancellationToken token(
+      [context]() mutable { context.CancelOperation(); });
+
+  auto hrn = hrn_;
+  auto client_settings = client_settings_;
+  data_request.WithLayerId(layer_id_);
```

Comment:
this line is not needed

Reply:
it is required to avoid to have an extra field in the class since the layer is present in the data request.